packages/xod-client/src/core/domSideeffectsMiddleware.js

```diff
@@ -0,0 +1,31 @@
+import waitForElement from 'wait-for-element';
+import { SELECT_CONSTANT_NODE_VALUE } from '../editor/actionTypes';
+
+export default _ => next => action => {
+  if (action.type === SELECT_CONSTANT_NODE_VALUE) {
+    // First of all, do the stuff in reducers to show inspector pane.
+    const n = next(action);
+    // Wait for updated Inspector, because it could appeared
+    // or it changed entirely on new selection
+    waitForElement('.Inspector-container')
+      .then(__ => {
+        // Select widget control with a tricky selector instead of ID, because
+        // some constant nodes has a generated id of the terminal Node
+        const ctrl = document.querySelector(
+          '.PinWidget input, .PinWidget select'
+        );
+        if (!ctrl) return;
+        if (ctrl.setSelectionRange)
+          ctrl.setSelectionRange(0, ctrl.value.length);
+        ctrl.focus();
+      })
+      .catch(err => {
+        // TODO: Add User-friendly error message?
+        throw err;
```

Comment:
This should never happen as I understand. So if it happens it is a bug. Just throwing the error here looks fine. Suggest removing TODO